content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,380 @@
+---
+title: Security Checklist
+description: >
+  Baseline checklist for ensuring security in Kubernetes clusters.
+content_type: concept
+---
+
+<!-- overview -->
+
+This checklist aims at providing a basic list of guidance with links to more
+comprehensive documentation on each topic. It does not claim to be exhaustive
+and is meant to evolve.
+
+The order of topics does not reflect an order of priority.
+
+{{< caution >}}
+Checklists are **not** sufficient for attaining a good security posture on their
+own. A good security posture requires constant attention and improvement, but a
+checklist can be the first step on the never-ending journey towards security
+preparedness. Some of the recommendations in this checklist may be too
+restrictive or too lax for your specific security needs. Since Kubernetes
+security is not "one size fits all", each category of checklist items should be
+evaluated on its merits.
+{{< /caution >}}
+
+<!-- body -->
+
+## Authentication & Authorization
+
+- [ ] `system:masters` group is not used for user or component authentication after bootstrapping.
+- [ ] The kube-controller-manager is running with `--use-service-account-credentials`
+  enabled.
+- [ ] The root certificate is protected (either an offline CA, or a managed
+  online CA with effective access controls).
+- [ ] Intermediate and leaf certificates have an expiry date no more than 3
+  years in the future.
+- [ ] A process exists for periodic access review, and reviews occur no more
+  than 24 months apart.
+
+See the [Role Based Access Control Good Practices](/docs/concepts/security/rbac-good-practices/)
+for guidance related to authentication and authorization.
+
+After bootstrapping, neither users nor components should authenticate to the
+Kubernetes API as `system:masters`. Similarly, running all of
+kube-controller-manager as `system:masters` should be avoided. In fact,
+`system:masters` should only be used as a break-glass mechanism, as opposed to
+an admin user.
+
+## Network security
+
+- [ ] CNI plugins in-use supports network policies.
+- [ ] Ingress and egress network policies are applied to all workloads in the
+  cluster.
+- [ ] Default network policies within each namespace, selecting all pods, denying
+  everything, are in place.
+- [ ] Use of a service mesh to encrypt all communications inside of the cluster.
```

Comment:
I don't think so. At the top of the page there is a bit of a disclaimer that states:
>Some of the recommendations in this checklist may be too
restrictive or too lax for your specific security needs. Since Kubernetes
security is not "one size fits all", each category of checklist items should be
evaluated on its merits.

Reply:
We could ship this as-is, and then file an issue to see if there's some even better way to word this particular item.

Reply:
I agree on this topic, and the other pointed out [here](https://github.com/kubernetes/website/pull/33992#discussion_r952810587). But I think the idea of putting the information about service mesh and secrets injectors was to highlight that this might be a sensitive part of the cluster that might require extra care in some situations. But indeed it's impossible to write a threat model that would suit every Kubernetes cluster usage, I will try to rephrase this with something like "consider using".

Reply:
```suggestion
- [ ] If appropriate, a service mesh is used to encrypt all communications inside of the cluster.
```